src/rai/rai/tools/ros/debugging.py

```diff
@@ -0,0 +1,199 @@
+            raise ValueError("Action name is required for info and type commands")
+
+    cmd = ["ros2", "action", command]
+    if action_name:
+        cmd.append(action_name)
+    return run_command(cmd, timeout)
+
+
+@tool
+def ros2_service(
+    command: Literal["call", "find", "info", "list", "type"],
+    service_name: str = "",
+    timeout: int = 5,
+):
+    """Run a ROS2 service command
+    Args:
+        command: The service command to run
+        service_name: Name of the service (required for call, info, and type)
+        timeout: Command timeout in seconds
+    """
+    if command in ["call", "info", "type"]:
+        if not service_name:
+            raise ValueError(
+                "Service name is required for call, info, and type commands"
+            )
+
+    cmd = ["ros2", "service", command]
+    if service_name:
+        cmd.append(service_name)
+    return run_command(cmd, timeout)
+
+
+@tool
+def ros2_node(command: Literal["info", "list"], node_name: str = "", timeout: int = 5):
+    """Run a ROS2 node command
+    Args:
+        command: The node command to run
+        node_name: Name of the node (required for info)
+        timeout: Command timeout in seconds
+    """
+    if command == "info":
+        if not node_name:
+            raise ValueError("Node name is required for info command")
+
+    cmd = ["ros2", "node", command]
+    if node_name:
+        cmd.append(node_name)
+    return run_command(cmd, timeout)
+
+
+@tool
+def ros2_param(
+    command: Literal["delete", "describe", "dump", "get", "list", "set"],
+    node_name: str = "",
+    param_name: str = "",
+    timeout: int = 5,
+):
+    """Run a ROS2 parameter command
+    Args:
+        command: The parameter command to run
+        node_name: Name of the node
+        param_name: Name of the parameter (required for get, set, delete)
+        timeout: Command timeout in seconds
+    """
+    if command in ["get", "set", "delete"]:
+        if not param_name:
+            raise ValueError(
+                "Parameter name is required for get, set, and delete commands"
+            )
+
+    cmd = ["ros2", "param", command]
+    if node_name:
+        cmd.append(node_name)
+    if param_name:
+        cmd.append(param_name)
+    return run_command(cmd, timeout)
+
+
+@tool
+def ros2_interface(
+    command: Literal["list", "package", "packages", "proto", "show"],
+    interface_name: str = "",
+    timeout: int = 5,
+):
+    """Run a ROS2 interface command
+    Args:
+        command: The interface command to run
+        interface_name: Name of the interface (required for show, proto)
+        timeout: Command timeout in seconds
+    """
+    if command in ["show", "proto"]:
+        if not interface_name:
+            raise ValueError("Interface name is required for show and proto commands")
+
+    cmd = ["ros2", "interface", command]
+    if interface_name:
+        cmd.append(interface_name)
+    return run_command(cmd, timeout)
```

Comment:
I think `ros2 interface package` might not be supported, because `package_name` param is required (maybe if LLM decides to use `interface_name` argument)
See:
```
robo-pc-005 ➜  12_rai git:(feat/debugging-tools) ✗ ros2 interface package --help
usage: ros2 interface package [-h] [-m] [-s] [-a] package_name
```

Reply:
I've loosen the restrictions, all the commands are available now